mmcv/runner/epoch_based_runner.py

```diff
@@ -107,7 +107,7 @@ def run(self, data_loaders, workflow, max_epochs=None, **kwargs):
                          self._max_epochs)
         self.call_hook('before_run')
 
-        while self.epoch < self._max_epochs:
+        while self.epoch < self._max_epochs and not self.should_stop:
```

Comment:
```suggestion
        while self.epoch < self._max_epochs:
             if self.should_stop:
                 self.logger.info('...')
                 break
```

Reply:
It is helpful for users if we print a proper log here.

Reply:
> Hi @24hours , thanks for your contribution.
> 
> There are two points to note:
> 
> * There are two ways to evaluate a dataset, one is to use [EvalHook](https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/evaluation.py) and the other is to set the workflow like `workflow=[('train', 1), ('val', 1)]`. We need to consider both of them.
> * The priority of hooks matters so we need to determine a reasonable priority for `EarlyStopHook`
>   https://github.com/open-mmlab/mmcv/blob/e4b5348ebf407f0886bc11caa7ca74d2fbf693ac/mmcv/runner/base_runner.py#L503-L531



> 

Evalhook is registered with `priority=LOW` is most algorithm. 
https://github.com/open-mmlab/mmclassification/blob/master/mmcls/apis/train.py#L196
https://github.com/open-mmlab/mmdetection/blob/master/mmdet/apis/train.py#L197
If we registery EarlyStopping to `VERY_LOW` then it would be possible to monitor EvalHook output. 

Since EarlyStoppingHook just monitor `runner.log_buffer.output[name]` it should work fine. 
https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/evaluation.py#L365
However, one would need to set `strict` flag to false. 

There are 2 issue need to clarify in this case. 
1. is `VERY_LOW` priority appropriate for EarlyStopping hook ?
    - I think it's ok because it doesnt seems to conflict with any other hook. 
    - However, it does feel weird that EShook is same rank with loggerhook. 
2. Should `strict` flag set to `false` by default ?
    - I opt for `strict = true` because strict pevent user giving errorneous log name. 
    - EShook should provide instruction to user upon validation check failed. Notify user to set `strict = false`